common/src/main/java/com/wynntils/features/user/QuickCastFeature.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyHolder;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wc.event.WorldStateEvent;
+import com.wynntils.wc.utils.WynnItemMatchers;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.network.protocol.Packet;
+import net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket;
+import net.minecraft.network.protocol.game.ServerboundSwingPacket;
+import net.minecraft.network.protocol.game.ServerboundUseItemPacket;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(stability = FeatureInfo.Stability.STABLE)
+public class QuickCastFeature extends UserFeature {
+
+    @RegisterKeyBind
+    private final KeyHolder castFirstSpell =
+            new KeyHolder("Cast 1st Spell", GLFW.GLFW_KEY_Z, "Wynntils", true, QuickCastFeature::castFirstSpell);
+
+    @RegisterKeyBind
+    private final KeyHolder castSecondSpell =
+            new KeyHolder("Cast 2nd Spell", GLFW.GLFW_KEY_X, "Wynntils", true, QuickCastFeature::castSecondSpell);
+
+    @RegisterKeyBind
+    private final KeyHolder castThirdSpell =
+            new KeyHolder("Cast 3rd Spell", GLFW.GLFW_KEY_C, "Wynntils", true, QuickCastFeature::castThirdSpell);
+
+    @RegisterKeyBind
+    private final KeyHolder castFourthSpell =
+            new KeyHolder("Cast 4th Spell", GLFW.GLFW_KEY_V, "Wynntils", true, QuickCastFeature::castFourthSpell);
+
+    private static final Pattern SPELL_PATTERN =
+            StringUtils.compileCCRegex("§([LR]|Right|Left)§-§([LR?]|Right|Left)§-§([LR?]|Right|Left)§");
+    private static final Pattern INCORRECT_CLASS_PATTERN = StringUtils.compileCCRegex("§✖§ Class Req: (.+)");
+    private static final Pattern LVL_MIN_NOT_REACHED_PATTERN = StringUtils.compileCCRegex("§✖§ (.+) Min: ([0-9]+)");
+
+    public static final SpellDirection[] NO_SPELL = new SpellDirection[0];
+    private static SpellDirection[] partialSpell = NO_SPELL;
```

Comment:
Could you either add a comment like "What we know about the current spell in progress", or perhaps rename it "spellInProgress"? Otherwise the purpose of this was a bit vague to me.

Reply:
That's a much bettern name, I'll change it when I get to a computer.